services/py-api/tests/integration_tests/conftest.py

```diff
@@ -82,3 +82,24 @@ async def _create(participant_body: Dict[str, Any]) -> Response:
         if result.status_code == 201:
             LOG.debug("Cleaning up test participant")
             await clean_up_test_participant(async_client=async_client, result_json=result.json())
+
+
+@pytest_asyncio.fixture
+def generate_participant_request_body() -> Callable[..., Dict[str, Any]]:
+    def _generate_participant_request_body(
+        name: str | None = "testtest",
+        email: str | None = "testtest@test.com",
+        is_admin: bool | None = False,
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+
+        request_body = {"name": name, "email": email, "is_admin": is_admin, **kwargs}
+        # A new dict without None values
+        return {key: value for key, value in request_body.items() if value is not None}
+
+    return _generate_participant_request_body
```

Comment:
I think it would make more sense to call the function `get_participant_request_body_generator` or something of the sorts

```python
generate_participant_request_body = get_participant_request_body_generator()
request_body = generate_participant_request_body("Misho", "notmyemail@gmail.com", True)
```

Reply:
I have used the same pattern as it is noted in the documentation of pytest fixtures. Here is a link to the docs: https://docs.pytest.org/en/6.2.x/fixture.html#factories-as-fixtures

Reply:
Functionality is correct. I'm not making a comment on this but rather the naming of the fixture.

Reply:
Oh I see that's right.